app/src/ui/pre-commit-hook-failure/pre-commit-hook-failure-dialog.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react'
+
+import { Repository } from '../../models/repository'
+import { Dispatcher } from '../dispatcher'
+import { ICommitContext } from '../../models/commit'
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { OkCancelButtonGroup } from '../dialog/ok-cancel-button-group'
+
+interface IPreCommitHookFailureProps {
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+  readonly commitContext: ICommitContext
+  readonly hookOutput: string
+  readonly onDismissed: () => void
+  readonly onCommitAnyway: () => void
+}
+
+interface IPreCommitHookFailureState {
+  /**
+   * Whether or not we're currently in the process of 
+   * committing with --no-verify. This is used to display a loading state
+   */
+  readonly isCommittingWithNoVerify: boolean
+}
+
+/** A component to handle pre-commit hook failures with bypass option. */
+export class PreCommitHookFailure extends React.Component<
+  IPreCommitHookFailureProps,
+  IPreCommitHookFailureState
+> {
+  public constructor(props: IPreCommitHookFailureProps) {
+    super(props)
+
+    this.state = {
+      isCommittingWithNoVerify: false,
+    }
+  }
+
+  public render() {
+    const title = __DARWIN__ ? 'Pre-commit Hook Failed' : 'Pre-commit hook failed'
+
+    return (
+      <Dialog
+        id="pre-commit-hook-failure"
+        type="error"
+        title={title}
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.onCommitAnyway}
+        loading={this.state.isCommittingWithNoVerify}
+        disabled={this.state.isCommittingWithNoVerify}
+        role="alertdialog"
+        ariaDescribedBy="pre-commit-hook-failure-description"
+      >
+        <DialogContent>
+          <div id="pre-commit-hook-failure-description" className="selectable-text">
+            <p>
+              The pre-commit hook failed to run successfully. The hook output is shown below:
+            </p>
+            <div className="hook-output">
+              <pre className="selectable-text">{this.props.hookOutput}</pre>
+            </div>
+            <p>
+              You can bypass the pre-commit hook by committing anyway, which will use the{' '}
+              <code>--no-verify</code> option. This is not recommended unless you are sure your changes are correct.
```

Comment:
I am open to different copy for this

Reply:
We'd rather hide the "git" details, so I think the `, which will use the --no-verify option` part should be removed.